### ISSUE TITLE
`allow(bad_style)` should silence `upper_case_acronyms`

### ISSUE THREAD

**Issue by nox** [rust-lang/rust-clippy#7041]
Allowing bad style should silence all style-related lints, including `upper_case_acronyms`.

```rust
#[allow(bad_style)]
struct BIO_METHOD(*mut ffi::BIO_METHOD);
```

**Comment by camsteffen** [comment]
`bad_style` is documented as a specific group of Rust built-in lints. So it would be surprising for it to affect Clippy lints. Though this could change if `upper_case_acronyms` is moved to rustc.

**Comment by nox** [comment]
Why would it be surprising? I don't follow.

**Comment by camsteffen** [comment]
Because upper_case_acronyms is not in the bad_style lint group.

**Comment by nox** [comment]
But it could very well include in its documentation that it is silenced when `bad_style` is allowed.

Currently, any new style-related clippy lint needs to be explicitly allowed even though there is the catch-all `bad_style` on them, that's way more surprising to me. I put `allow(bad_style)` somewhere therefore I should not see style-related warnings.

**Comment by camsteffen** [comment]
Clippy lints are intentionally kept separate from Rustc built-in lints because they are not as stable. And Rustc documentation does not mention Clippy because it is a separate tool. You should adjust your expectations for `bad_style`. Only lint names and lint groups prefixed with `clippy::` will affect Clippy lints.

**Comment by nox** [comment]
I don't see why you keep bringing the fact that they are separate. They are indeed separate but that lint in particular is a *style* lint, so I don't see why it shouldn't be silenced by `bad_style`. To me your reasoning sounds circular. You closed the issue though so I won't keep arguing about it.

**Comment by llogiq** [comment]
Does `clippy::style` != `bad_style` sound like circular reasoning?

**Comment by nox** [comment]
I see it as circular reasoning because it sounds like you are saying "clippy lints won't interact with libstd lints because clippy lints won't interact with libstd lints". It feels like you are prescriptively rejecting the idea before even considering it.

I understand that there is no precedent for clippy lints to be silenced by standard lints, but it seems early to me to pretend that creating such a precedent would be surprising or unexpected by end users.

What I feel is unexpected, personally, is the fact that any new warn-by-default style-related clippy lint cannot be silenced by any catch-all lint, and I feel like such a catch-all lint should be `bad_style`, cause that's what, say, people doing FFI use.

To me it seems unfortunate that there is no way to prevent any style-related lint to be issued even in the future, for stuff on which we don't control the style at all, and that's what FFI is.
